qlib/linux_def.rs

```diff
@@ -3039,6 +3044,193 @@ impl MemoryDef {
+    pub const fn phy_lower_gpa () -> u64 {
+        return Self::PHY_LOWER_ADDR;
+    }
+
+    pub const fn phy_lower_hva () -> u64 {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "cc")] {
+                if ENABLE_EMULATION_CC {
+                    return Self::PHY_LOWER_ADDR - Self::ONE_GB;
+                } else {
+                    return Self::PHY_LOWER_ADDR;
+                }
+            } else {
+                return Self::PHY_LOWER_ADDR;
+            }
+        }
+    }
+
+    pub const fn guest_private_init_heap_offset_gpa () -> u64 {
+        return Self::GUEST_PRIVATE_HEAP_OFFSET;
+    }
+
+    pub const fn guest_private_init_heap_offset_hva () -> u64 {
+
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "cc")] {
+                if ENABLE_EMULATION_CC {
+                    return Self::GUEST_PRIVATE_HEAP_OFFSET - Self::ONE_GB;
+                } else {
+                    return Self::GUEST_PRIVATE_HEAP_OFFSET;
+                }
+            } else {
+                return Self::GUEST_PRIVATE_HEAP_OFFSET;
+            }
+        }
+    }
+
+    pub const fn guest_private_init_heap_end_gpa () -> u64 {
+        return Self::GUEST_PRIVATE_INIT_HEAP_SIZE + Self::GUEST_PRIVATE_HEAP_OFFSET;
+    }
+
+    pub const fn guest_private_init_heap_end_hva () -> u64 {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "cc")] {
+                if ENABLE_EMULATION_CC {
+                    return Self::GUEST_PRIVATE_INIT_HEAP_SIZE + Self::GUEST_PRIVATE_HEAP_OFFSET - Self::ONE_GB;
+                } else {
+                    return Self::GUEST_PRIVATE_INIT_HEAP_SIZE + Self::GUEST_PRIVATE_HEAP_OFFSET;
+                }
+            } else {
+                return Self::GUEST_PRIVATE_INIT_HEAP_SIZE + Self::GUEST_PRIVATE_HEAP_OFFSET;
+            }
+        }
+    }
+
+    pub const fn guest_private_init_heap_size () -> u64 {
+        Self::GUEST_PRIVATE_INIT_HEAP_SIZE
+    }
+
+
+    pub const fn guest_private_running_heap_size () -> u64 {
+        return Self::GUEST_PRIVATE_HEAP_SIZE - Self::GUEST_PRIVATE_INIT_HEAP_SIZE; 
+    }
+
+    pub const fn guest_private_running_heap_offset_gpa () -> u64 {
+        return Self::GUEST_PRIVATE_HEAP_OFFSET + Self::GUEST_PRIVATE_INIT_HEAP_SIZE; 
+    }
+
+
+    pub const fn guest_private_running_heap_end_gpa () -> u64 {
+        return Self::GUEST_PRIVATE_HEAP_END;
+    }
+
+    pub const fn rdma_local_shared_gpa () -> u64 {
+        return Self::RDMA_LOCAL_SHARE_OFFSET;
+    }
+
+
+    pub const fn rdma_local_shared_hva () -> u64 {
+        return Self::RDMA_LOCAL_SHARE_OFFSET;
+    }
+
+
+    pub const fn rdma_global_shared_gpa () -> u64 {
+        return Self::RDMA_GLOBAL_SHARE_OFFSET;
+    }
+
+
+    pub const fn rdma_global_shared_hva () -> u64 {
+        return Self::RDMA_GLOBAL_SHARE_OFFSET;
+    }
+
+    //======================== shared ====================//
+
+    pub const fn phy_upper_gpa () -> u64 {
```

Comment:
if these stay constant and there is no runtime calculation, why do we need getter functions?

Reply:
IMHO having getters is not solving the real issue, if not making it worse. Because all the defs constant are still in the same place i.e. MemDefs. If separating host and guest memory defs is the case then we should consider refactoring MemoryDefs first.